### PR TITLE
Double-click word selection for RichTextLabel (i.e. docs)

### PR DESCRIPTION
**Pull request by ghost**
I.e.:

words

### CODE REVIEW

File: scene/gui/rich_text_label.cpp [hunk 1]
```diff
@@ -815,7 +842,36 @@ void RichTextLabel::_gui_input(Ref<InputEvent> p_event) {
 							}
 						}
 					}
+				} else if (b->is_pressed() && b->is_doubleclick() && selection.enabled) {
 
+					//doubleclick: select world
```
Comment (Noshyaar) [inline scene/gui/rich_text_label.cpp]: Not sure if I want to select a world in my RichTextLabel. :P

Comment (poke1024) [reply scene/gui/rich_text_label.cpp]: Well, sometimes a word's a world :-) Indeed I copied this over from `TextEdit`. Corrected for both comments now.

File: scene/gui/rich_text_label.cpp [hunk 2]
```diff
@@ -36,6 +36,33 @@
 #include "editor/editor_node.h"
 #endif
 
+static bool _is_symbol(CharType c) {
+	return c != '_' && ((c >= '!' && c <= '/') || (c >= ':' && c <= '@') || (c >= '[' && c <= '`') || (c >= '{' && c <= '~') || c == '\t' || c == ' ');
```
Comment (Noshyaar) [inline scene/gui/rich_text_label.cpp]: I'd put this alongside similar functions in ustring.cpp just to keep similar stuff together.
https://github.com/godotengine/godot/blob/5c636875e418fbf055ddcff8d682639d1f096d05/core/ustring.cpp#L54-L57

Comment (poke1024) [reply scene/gui/rich_text_label.cpp]: Moved `is_symbol` to `ustring`.

Also deduplicated `select_word` and moved it to `ustring` as well.

**Pull request merged**